resources/aws_ec2.go

```diff
@@ -972,6 +1023,84 @@ func (obj *AwsEc2Res) snsConfirmSubscription(topicArn string, token string) erro
 	return nil
 }
 
+// cweMakeEventPattern makes and encodes event patterns for cloudwatch rules.
+func (obj *AwsEc2Res) cweMakeEventPattern(source string, detailType string, detail []string) (string, error) {
+	pattern := cloudWatchRule{
+		Source:     []string{source},
+		DetailType: []string{detailType},
+		Detail: ruleDetail{
+			State: detail,
+		},
+	}
+	eventPattern, err := json.Marshal(pattern)
+	if err != nil {
+		return "", err
+	}
+	return string(eventPattern), nil
+}
+
+// cweMakeRule makes a cloud watch rule.
+func (obj *AwsEc2Res) cweMakeRule(name string, eventPattern string) error {
+	// make cloudwatch rule
+	putRuleInput := &cwe.PutRuleInput{
+		Name:         aws.String(name),
+		EventPattern: aws.String(eventPattern),
+	}
+	_, err := obj.cweClient.PutRule(putRuleInput)
+	if err != nil {
+		return err
+	}
+	log.Printf("%s: Created CloudWatch Rule", obj)
+	return nil
+}
+
+// cweDeleteRule deletes the cloudwatch rule.
+func (obj *AwsEc2Res) cweDeleteRule(name string) error {
+	// delete the rule
+	drInput := &cwe.DeleteRuleInput{
+		Name: aws.String(name),
+	}
+	log.Printf("%s: Deleting CloudWatch Rule", obj)
+	if _, err := obj.cweClient.DeleteRule(drInput); err != nil {
```

Comment:
No need to change, this is fine (and what I would have expected), however FYI as a fun learning exercise, the following code is actually equivalent! Do you know why?

```
	err := obj.cweClient.DeleteRule(drInput)
	return errwrap.Wrapf(err, "error deleting cloudwatch rule")
```

Reply:
I haven't seen (maybe I just haven't noticed) this pattern before. This doesn't result in a compiler error since there are two return values and we're only assigning one?

Reply:
Woops, typo. Fixed above. Check again...

Reply:
@purpleidea That would return nil if there was no error, which I do right after. It would save 3 LoC, but I like the explicit-ness of having both returns. Good trick to have in the tool-bag though.